camp/settings.py

```diff
@@ -537,13 +470,36 @@
 # for now we remove CsrfViewMiddleware, which creates failures on x-csrftoken
 # We need to fix this
 
-# MIDDLEWARE_CLASSES += (
-#     # 'django.middleware.csrf.CsrfViewMiddleware', # remove csrf because it will stop users from extranet
-#     # Add cache middleware for the per-site cache
-#     'django.middleware.cache.UpdateCacheMiddleware',
-#     'django.middleware.common.CommonMiddleware',
-#     'django.middleware.cache.FetchFromCacheMiddleware',
-# )
+MIDDLEWARE_CLASSES = (
+    'corsheaders.middleware.CorsMiddleware',
+    'django.middleware.common.CommonMiddleware',
+    'django.contrib.sessions.middleware.SessionMiddleware',
+    'django.contrib.messages.middleware.MessageMiddleware',
+    'dj_pagination.middleware.PaginationMiddleware',
+    # The setting below makes it possible to serve different languages per
+    # user depending on things like headers in HTTP requests.
+    'django.middleware.locale.LocaleMiddleware',
+    # 'django.middleware.csrf.CsrfViewMiddleware',
```

Comment:
I will merge this if you really want, but keep in mind that this is highly insecure!
Why do you need it? If something doesn't work well with feature editing or style editing is most likely caused by misconfiguration. In staging everything works well with CsrfViewMiddleware enabled. And also in Worldmap

Reply:
Because the camp server is hosted in Zhejiang University now. The users out of the inner-network of Zhejiang University will meet with the error of csrf. So we should close it in camp server.